sysl2/sysl/grammar/SyslParser.g4

```diff
@@ -42,6 +42,10 @@ table   :   SYSL_COMMENT*
             (TABLE | TYPE)
             name_str attribs_or_modifiers? COLON ( WHATEVER | INDENT (SYSL_COMMENT | field | annotation | inplace_table | WHATEVER )+ DEDENT)
         ;
+union   :   SYSL_COMMENT*
+            UNION
+            name_str attribs_or_modifiers? COLON ( WHATEVER | INDENT (SYSL_COMMENT | user_defined_type | annotation | inplace_table | WHATEVER )+ DEDENT)
+        ;
```

Comment:
just like `table` has a `field`, what does union have? 
It would be great if you could define something similar for union with the added benefit of being able to reuse user_defined_type. 

Eg rule (abbreviated) and its implementation:
union_element: user_defined_type | inplace_table;
union: UNION name_str COLON 
                 INDENT ( annotation | union_element )+ DEDENT;

Now, in the listener impl:
In `EnterUnion_Element`: you can append to fieldname and re-use the implementation of ExitUser_defined_type.

Reply:
Another idea to explore: you can use the grammar to enforce that union should atleast have 2 elements.

Reply:
> could you please add working example of `inplace_table` ?

Not for MVP. I've removed the `inplace_table` term.